Classes/FairPlayDRM/FPSLicenseHelper.swift

```diff
@@ -0,0 +1,252 @@
+        
+        let done: (Error?) -> Void = { [unowned self] error in 
+            callback(error)
+            self.doneCallback?(error)
+        }
+        
+        let assetId = self.assetId
+        
+        if let store = self.dataStore {
+            if !forceDownload && store.fpskeyExists(assetId) {
+                do {
+                    let license = try store.loadFpsKey(assetId)
+                    if !license.isExpired() {
+                        request.processContentKeyResponse(license.data)
+                        done(nil)
+                    }
+                    
+                } catch {
+                    request.processContentKeyResponseError(error)
+                    done(error)
+                }
+                return
+            }
+        }
+        
+        
+        guard let params = FPSParams(self.params) else { done(FPSError.missingDRMParams); return }
+        let shouldPersist = self.shouldPersist
+        let dataStore = self.dataStore
+        
+        request.getSPC(cert: params.cert, id: assetId, shouldPersist: shouldPersist) { [weak self] (spcData, error) in                                                                
+            
+            guard let strongSelf = self else { return }
+            if let error = error {
+                request.processContentKeyResponseError(error)
+                done(error)
+                return
+            }
+            
+            guard let spcData = spcData else { return }
+            
+            // Send SPC to Key Server and obtain CKC
+            strongSelf.performCKCRequest(spcData, url: params.url) { (license, error) in 
+                guard let license = license else {
+                    request.processContentKeyResponseError(error!)
+                    done(error)
+                    return
+                }
+                                
+                if shouldPersist {
+                    do {
+                        let pck = try request.persistableContentKey(fromKeyVendorResponse: license.data, options: nil)
+                        license.data = pck
+                        
+                        try dataStore?.saveFpsKey(assetId, license)
+                        
+                    } catch {
+                        request.processContentKeyResponseError(error)
+                        done(error)
+                        return
+                    }
+                }
+                
+                request.processContentKeyResponse(license.data)
+                
+                done(nil)
+            }
+        }
+    }
+}
+
+struct FPSParams {
+    let cert: Data
+    let url: URL
+    init?(_ pkParams: FairPlayDRMParams?) {
+        guard let params = pkParams else { return nil }
+        guard let cert = params.fpsCertificate else { PKLog.error("Missing FPS certificate"); return nil }
+        guard let url = params.licenseUri else { PKLog.error("Missing FPS license URL"); return nil }
```

Comment:
Let's break the lines because there is more than just return.

Reply:
Same